Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java

```diff
@@ -50,31 +56,23 @@ public Connection(InputAnchor to, OutputAnchor from) {
     }
 
     /**
-     * Sets the free ends (empty anchors) to the specified position
-     */
-    public void setFreeEnds(double x, double y) {
-        if (!startAnchor.isPresent()) {
-            setStartPosition(x, y);
-        }
-        if (!endAnchor.isPresent()) {
-            setEndPosition(x, y);
-        }
-    }
-
-    /**
      * Tries to add an unspecified ConnectionAnchor to the connection.
      *
-     * @param anchor Anchor to add
-     * @param override If set will override (possible) existing Anchor.
+     * @param anchor
+     *            Anchor to add
+     * @param override
+     *            If set will override (possible) existing Anchor.
      * @return Whether or not the anchor was added.
      */
     public boolean addAnchor(ConnectionAnchor anchor, boolean override) {
         boolean added = false;
-        if ((!startAnchor.isPresent() || override) && anchor instanceof OutputAnchor) {
+        if ((!startAnchor.isPresent() || override)
+                && anchor instanceof OutputAnchor) {
```

Comment:
Yeah, it's obvious enough, but not really more obvious than the old version, I'd say.

How about something like this?

````
boolean added = false,
        isOutputAnchor = anchor instanceof OutputAnchor,
        isInputAnchor = anchor instanceof InputAnchor;

if (isOutputAnchor && (!startAnchor.isPresent() || override)) {
    disconnect(startAnchor);
...
````